PlanetWorker/src/PlanetWorker.cs

```diff
@@ -56,19 +67,140 @@ static int Main(string[] arguments)
+                entity.entity.Update<PlanetInfo>(update);
+            }
+          });
+          
+          dispatcher.OnAddEntity(op =>
+          {
+            var logMessage = String.Format("Adding entityId {0}", op.EntityId);
+            connection.SendLogMessage(LogLevel.Info, LoggerName, logMessage);
+
+            // AddEntity will always be followed by OnAddComponent
+            ViewEntity newEntity = new ViewEntity();
+            newEntity.hasAuthority = true;
+            newEntity.entity = new Entity();
+            ViewEntity oldEntity;
+            if(!EntityView.TryGetValue(op.EntityId, out oldEntity))
+            {
+              EntityView.Add(op.EntityId, newEntity);
+            }
+          });
+          
+          dispatcher.OnRemoveEntity(op =>
+          {
+            EntityView.Remove(op.EntityId);
+          });
 
-          // dispatcher.OnCommandRequest<PingResponder.Commands.Ping>(request =>
-          // {
-          //     connection.SendLogMessage(LogLevel.Info, LoggerName, "Received GetWorkerType command");
-          //
-          //     var greeting = hellos[random.Next(hellos.Length)];
-          //     var pingResponse = new Pong(WorkerType, String.Format("{0}, World!", greeting));
-          //     var commandResponse = new PingResponder.Commands.Ping.Response(pingResponse);
-          //     connection.SendCommandResponse(request.RequestId, commandResponse);
-          // });
+          dispatcher.OnCommandRequest<AssignPlanetResponder.Commands.AssignPlanet>(request =>
+          {
+            var logMessage = String.Format("Received AssignPlanet command from player {0}", request.Request.Get().Value.playerId);
+            connection.SendLogMessage(LogLevel.Info, LoggerName, logMessage);
+              
+              foreach (KeyValuePair<EntityId, ViewEntity> pair in EntityView)
+              {
+                if(!pair.Value.hasAuthority)
+                {
+                  logMessage = String.Format("Skipping planet with entityId {0} because I don't have authority", pair.Key);
+                  connection.SendLogMessage(LogLevel.Info, LoggerName, logMessage);
+                  continue;
+                }
+                
+                logMessage = String.Format("Picked planet with entityId {0} because I have authority", pair.Key);
+                connection.SendLogMessage(LogLevel.Info, LoggerName, logMessage);
+                
+                logMessage = String.Format("Entity.PlanetInfo: {0}", pair.Value.entity.Get<PlanetInfo>());
+                connection.SendLogMessage(LogLevel.Info, LoggerName, logMessage);
+                
+                Option<IComponentData<PlanetInfo>> option;
+                IComponentData<PlanetInfo> planetInfo;
+                PlanetInfo.Data planetInfoData;
+                PlanetInfoData pid;
+
+                option = pair.Value.entity.Get<PlanetInfo>();
+                planetInfo = option.Value;
```

Comment:
This may cause some bugs when you have entities that won't have planetinfo components on them